### PR TITLE
Added circular progress to various modals

### LINKED ISSUE #3319: Use Circular Progress wherever required
**Expected Behaviour**

Use `Circular Progress` wherever needed.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Dilshaad21**
Fixes #3319 

Changes: 
- Used `Circular Progress` in Modals while loading.
- Set a common `Button` ui to all modals.

Demo Link: https://pr-3327-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

**Comment by Dilshaad21** [comment]
@akshatnitd Made `LoadingButton` component and replaced it with previous Button

File: src/components/shared/LoadingButton.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        loading: true,
+      },
+      () => {
+        this.props.onConfirm();
+      },
+    );
+  };
+
+  render() {
+    const { loading } = this.state;
```
Comment (akshatnitd) [inline src/components/shared/LoadingButton.js]: No need to keep loading in the state. Assume it to be coming in props. And also let the name be `Button`.

Comment (Dilshaad21) [reply src/components/shared/LoadingButton.js]: Should I incorporate simple Button and LoadingButton in one file itself

Comment (akshatnitd) [reply src/components/shared/LoadingButton.js]: Assume, the default value of loading to be false. Then single component would work.

**Comment by Dilshaad21** [comment]
@akshatnitd I have made changes in all files where shared Button was used. And kept normal and loading buttons in one single component. Also, tell me if some changes need to be made in the naming of props of `Button` or anywhere else.

File: src/components/shared/Button.js [hunk 2]
```diff
@@ -1,8 +1,60 @@
...
+  const {
+    buttonText,
+    color,
+    variant,
+    isLoading,
+    isDisabled,
+    handleClick,
+    label,
+    style,
+  } = props;
```
Comment (akshatnitd) [inline src/components/shared/Button.js]: Use `const {loading, ...otherProps} = props;`

File: src/components/shared/Button.js [hunk 3]
```diff
@@ -1,8 +1,60 @@
...
+  return (
+    <StyledButton
+      variant={variant}
+      color={color}
+      onClick={handleClick}
+      disabled={isLoading || isDisabled}
+      label={label}
+      style={style}
```
Comment (akshatnitd) [inline src/components/shared/Button.js]: Pass `{...otherProps}` to the component here. This way we can ensure all props provided by MUI can be passed down to the button

**Comment by Dilshaad21** [comment]
@akshatnitd made changes, passed mui props using `otherProps`.

File: src/components/shared/Button.js [hunk 4]
```diff
@@ -1,8 +1,47 @@
...
+    isLoading,
+    isDisabled,
+    handleClick,
+    ...otherProps
+  } = props;
+  return (
+    <StyledButton
+      onClick={handleClick}
+      disabled={isLoading || isDisabled}
+      {...otherProps}
```
Comment (akshatnitd) [inline src/components/shared/Button.js]: `disabled` is already a Mui prop. You should use that instead of `isDisabled`.

**Comment by Dilshaad21** [comment]
@akshatnitd changes are done!

**Pull request merged**